test/navigation/imu_preprocessing.cpp

```diff
@@ -0,0 +1,60 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <navigation/preprocess_imu.hpp>
+#include <utils/manual_time.hpp>
+
+namespace hyped::test {
+
+core::Float epsilon = 1e-5;
+bool checkArrayEquality(const core::ImuData &imu_data_a, const core::ImuData &imu_data_b)
+{
+  if (imu_data_a.size() == imu_data_b.size()) {
+    for (std::size_t i; i < imu_data_a.size(); ++i) {
+      if (!(std::abs(imu_data_a.at(i) - imu_data_b.at(i)) < epsilon)) { return false; }
+    }
```

Comment:
this can be restructured to use guards instead of nesting it so much.

ie `if data_a != data_b return false`
then the for loop